src/structure.js

```diff
@@ -68,6 +69,34 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path, fn) {
+  var self = this;
+  fn = fn || function () { };
+
+  var cursor = this.cursor(path);
+  var changeListener = function (newRoot, oldRoot, changedPath) {
+    fn(newRoot, oldRoot, path);
+    cursor = self.cursor(path);
+  };
+  this.on('swap', changeListener);
+
+  return {
+    onChange: function (newFn) {
+      var original = fn;
+      fn = function () {
+        newFn.apply(this, arguments);
+        return original.apply(this, arguments);
+      }
+    },
+    cursor: function () {
```

Comment:
If we're going to have this, a dual `structure` should be provided to get the `structure` reference.